### PR TITLE
Answer comments frontend

### PR DESCRIPTION
**Pull request by logtheman**
In this PR:
- Adjusted Answer controller to be able to edit answers properly
- Added comment controller routing and inheritance for Question comments and Answer comments
- Plugged in Edit Answer and like / dislike answers 
- Plugged in commenting for both answers and questions
- Fixed commenting to be able to be edited and delete
- Added view count and auto increment

TODO: 
 - fix redirects (consider if necessary since we are moving to React)
 - Adjust styles and visual presentation for show comments, questions and answers

### CODE REVIEW

File: app/controllers/answers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Answers::CommentsController < CommentsController
```
Comment (jerflan) [inline app/controllers/answers/comments_controller.rb]: Are we not allowing comments on questions as well?

File: app/controllers/answers_controller.rb [hunk 2]
```diff
@@ -1,62 +1,76 @@
...
+  def destroy
+    @answer = Answer.find(params[:id])
+    @answer.destroy
+   
+    redirect_to :back
+  end
+
   def like
-    @answer.liked_by @user
+    @answer.liked_by current_user
```
Comment (jerflan) [inline app/controllers/answers_controller.rb]: Yay! Thank you! Have I mentioned how much I LOVE `current_user`? It's literally the best thing ever.

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
+class CommentsController < ApplicationController
```
Comment (jerflan) [inline app/controllers/comments_controller.rb]: I swear I look up singular/plural every time I generate a controller. Sorry about that.

File: db/migrate/20170729203107_add_views_count_to_questions.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddViewsCountToQuestions < ActiveRecord::Migration
```
Comment (jerflan) [inline db/migrate/20170729203107_add_views_count_to_questions.rb]: Have you built out the views_count functionality yet? If not, I wouldn't mind taking a stab at it.

File: app/controllers/questions/comments_controller.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+class Questions::CommentsController < CommentsController
```
Comment (logtheman) [inline app/controllers/questions/comments_controller.rb]: @jerflan - We have a comment controller for Questions as well. The generalized comments controller handles most everything though

File: app/controllers/answers_controller.rb (hunk 2 above)
Comment (logtheman) [inline app/controllers/answers_controller.rb]: @jerflan - no really, like amazing!

**Comment by logtheman** [comment]
@jerflan - Please go ahead and merge your PRs and I will sync them before merging this one in.

**Comment by jerflan** [comment]
Also, if the redirects are causing major testing issues, we should fix them, but if you're starting on a separate react client, then i wouldn't worry about it.

I can help work on the views-count and finish the tagging once you merge this in.

**Comment by jerflan** [comment]
What was the reasoning behind a separate comments for questions controller?

**Comment by logtheman** [comment]
Redirects are not major problems and work for testing for now. They just don't make sense and would be a not amazing user experience if left in the current state (obviously we won't). Like saving a comment and going back to the question index. I'm not married to working on it because it will become irrelevant when we use react and change the state without reloading. 

So, they are definitely more than one way to do the comments. This is just one of the ways I had read about. Basically the comment controller does most the lifting, but there is a questions and answers comments controllers that inherit. This way you can have specific methods (such as set commentable) that will apply for questions or answers. Probably not strictly necessary to do it this way, but I thought it made sense.

**Pull request merged**